src/main/scala/com/github/jberggg/router/RouterDsl.scala

```diff
@@ -0,0 +1,33 @@
+package com.github.jberggg.router
+
+import cats.implicits._
+import cats.effect.Async
+import org.scalajs.dom.window
+import fs2.concurrent.Channel
+import fs2.Stream
+import org.http4s.Uri.Path
+
+import Domain._
+
+trait RouterDsl[F[_]] {
+
+    def requestedPaths: Stream[F, Path]
```

Comment:
Same comment here, I actually think this is better represented as a `Signal`. Since a `Signal` models both a "right now" and a stream of events.

Reply:
I was thinking this one might become a `Signal` as well, since it's user-facing API :)

Reply:
Sorry, totally over-read that comment and just closed it. As you maybe can tell I'm new-ish to fs2 and actually run into "difficulties" using Signal because I first created the stream with `continuous` which lead to the behavior that current path continued to pop up. Once I changed to `discrete` behavior was a expected (by me). Probably had to due with the consumer (outwatch app) and if I'm more skilled in fs2 I would have known how to create the stream from the beginning. But it just made me wondering, although it is semantically nicer, wouldn't a Stream may be a simpler more fool proof API for  people like me? 😂

Reply:
Sure, totally up to you, it's your project! :)

Besides the semantics, a `Signal` essentially offers two advantages over `Stream`:
1. You can always query it for the current location, without have to wait for the next location to be emitted (as with `Stream`).
2. You can have multiple subscribers to a `Signal`, without "re-creating" the original source. Meanwhile, every time you `.compile` a `Stream` it may involve several side-effects (which in this case would be setting up a listener and dispatcher).

Reply:
Well of course I could do "my thing" but I don't like that thinking and this mini-library should be as useful to the community as possible :)

Hmm yes I agree, Signal gives more freedom to work with. I will change it 👍🏽